controllers/index.js

```diff
@@ -0,0 +1,317 @@
+					}
+					else {
+						res.attachment(file.file_name);
+						res.send(data.Body);	
+					}
+
+				});
+			}
+			else {
+				error.sendError(req, res, 400, 'Invalid controlNumber or fileName provided');
+			}
+		}
+	});
+	
+};
+
+const getControlNumber = function(req, res, reqData){
+	const pathData = reqData.schema;
+	const fileTypes = {
+		'gud': 'guideDocumentation',
+		'arf': 'acknowledgementOfRiskForm',
+		'inc': 'insuranceCertificate',
+		'gse': 'goodStandingEvidence',
+		'opp': 'operatingPlan'
+	};
+
+	const basicData = getBasicRes(pathData);
+
+	let jsonData = {};
+
+	const controlNumber = reqData.matches.controlNumber;
+
+	const jsonResponse = {};
+	jsonResponse.success = true;
+	jsonResponse.api = 'FS ePermit API';
+	jsonResponse.type = 'controller';
+	jsonResponse.verb = req.method;
+	jsonResponse.src = 'json';
+	jsonResponse.route = req.originalUrl;
+
+	const cnData = basicData[1095010356];  // TODO: remove - used for mocks
+
+	if (basicData){
+		db.getApplication(controlNumber, function(err, appl, fileData){
+			if (err){
+				return error.sendError(req, res, 400, 'error getting application from database');
+			}
+			else {
+				if (fileData){
+					fileData.forEach(function(file){
+						const fileType = fileTypes[file.file_type];
+						appl[fileType] = file.file_name;
+					});
+				}
+				jsonData = get.copyGenericInfo(cnData, appl, jsonData, pathData.getTemplate);
+				jsonData.controlNumber = controlNumber;// TODO: remove - used for mocks
+				const toReturn = Object.assign({}, {response:jsonResponse}, jsonData);
+
+				res.json(toReturn);
+			}
+		});
+	}
+
+};
+
+//*************************************************************
+
+const postApplication = function(req, res, reqData){
+	const pathData = reqData.schema;
+
+	const body = getBody(req);
+	const derefFunc = deref();
+	const possbileFiles = [];
+
+	const schema = validation.getValidationSchema(pathData);
+	const sch = derefFunc(schema.schemaToUse, [schema.fullSchema]);
+	const allErrors = validation.getFieldValidationErrors(body, pathData, sch);
+	
+	//Files to validate are in possbileFiles
+	validation.checkForFilesInSchema(sch, possbileFiles);
+
+	if (possbileFiles.length !== 0){
+		possbileFiles.forEach((fileConstraints)=>{
+			const key = Object.keys(fileConstraints)[0];
+			const fileValidationErrors = validation.validateFile(req.files[key], fileConstraints, key);
+			allErrors.errorArray = allErrors.errorArray.concat(fileValidationErrors);
+		});
+	}
+	const errorMessage = validation.generateErrorMesage(allErrors);
+	if (allErrors.errorArray.length !== 0){
+		return error.sendError(req, res, 400, errorMessage, allErrors.errorArray);
+	}
+	else {
+		const toStoreInDB = db.getDataToStoreInDB(sch, body);
+
+		const controlNumber = (Math.floor((Math.random() * 10000000000) + 1)).toString(); //TODO: remove - used for mocks
+		toStoreInDB.control_number = controlNumber;
+		db.saveApplication(controlNumber, toStoreInDB, function(err, appl){
```

Comment:
This order of saving application and calling basic API needs to be swapped, can be done in new sub task.